server/pbenchinacan/pod.yml

```diff
@@ -75,6 +78,16 @@ spec:
       - name: vm.max_map_count
         value: "262144"
     workingDir: /usr/share/elasticsearch
+  - command:
+    - /opt/keycloak/bin/kc.sh
+    - start-dev
+    - --health-enabled=true
```

Comment:
Looking at the [documentation reference](https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/health.adoc) that you pointed us to, it's not obvious to me that adding `--health-enabled` changes anything.

The doc says
> By default, no check is returned from the health endpoints.

which I interpret to mean that the `checks` field in the JSON body of the response object will always be `[]`, by default.  But, I expect that the API still returns the `"status"` field, even if we don't enable the additional check(s).

The doc says (emphasis mine)
> It is possible to enable the health checks using the **_build time_** option `health-enabled`:

I don't know what they mean by "build time", but I would assume that what you're specifying at line 84 is a **_run_** time option.

Also, according to the doc, `health-enabled` would enable the one and only available check except that it _also_ requires using the `metrics-enabled` option.

So, I don't think that line 84 is actually accomplishing anything.  (I recommend removing it and seeing if everything still works satisfactorily.)

Reply:
We are starting the Keycloak server with `start-dev` here and there is no `build` stage for the start-dev operation. It is applicable when we don't use the dev Keycloak container and want to build the image first so that that the start time is not compromised. 


> Also, according to the doc, `health-enabled` would enable the one and only available check except that it _also_ requires using the `metrics-enabled` option.

`Checks` field in the JSON body of the response object will always be [], that is correct. However, I am not enabling any checks in this PR, what I want to check here is that whether the Keycloak server is up or not to use. Enabling the metrics is not necessary here because we are using the dev Keycloak server which uses internal database other configurations are default. 

> So, I don't think that line 84 is actually accomplishing anything. (I recommend removing it and seeing if everything still works satisfactorily.)

It checks whether the server is up or not and that's all I want to check for now. Maybe I will enable other metrics later when we start using the `non-dev` Keycloak server.

Reply:
@npalaska and I chatted and reached the following conclusions:
- with the `start-dev` command, the Keycloak instance is built on the fly, so the build options are meaningful
- without the `health-enabled` option, the Keycloak server actually doesn't export the `/health` endpoint
- and so, for our customized, containerized Keycloak instance, we do need the `health-enabled` option
- but, since all we're interested in is the base status, we don't need the `metrics-enabled` option